pipenv/project.py

```diff
@@ -685,8 +685,8 @@ def create_pipfile(self, python=None):
         from .vendor.pip_shims.shims import (
             ConfigOptionParser, make_option_group, index_group
         )
-
-        config_parser = ConfigOptionParser(name=self.name)
+        # Inherit the pip's index configuration of install command.
+        config_parser = ConfigOptionParser(name="install")
```

Comment:
Arguably, we should change this to `pip_shims.shims.InstallCommand()` and grab the relevant option groups from there. Trying to construct this ourselves out of option groups is foolish and I don't think it actually works correctly.

```suggestion
        command = InstallCommand()
        extra_indexes = command.cmd_opts.get_option("--extra-index-url").default
```

```python
>>> from pip_shims.shims import InstallCommand
>>> cmd = InstallCommand()
>>> cmd.cmd_opts.get_option("--extra-index-url").default
[]
```